poller/poller.go

```diff
@@ -34,6 +42,20 @@ type SplitData struct {
 	UsingSegmentsCount int
 }
 
+// Cache contains raw split data as well as the data in serialized format
+type Cache struct {
```

Comment:
nit: can we put Cache before SplitData, putting SplitData and SplitCachePreload together.